packages/grpc-native-core/index.d.ts

```diff
@@ -193,7 +195,7 @@ declare module "grpc" {
      * @param implementation Map of method names to method implementation
      * for the provided service.
      */
-    addService(service: Service, implementation: { [name: string]: handleCall }): void;
+    addService<T>(service: Service<T>, implementation: T): void;
```

Comment:
This seems to make the function signature less useful by removing the specific definition of the type of `implementation`, and I don't see any clear benefit. Because the `Service<T>` definition doesn't even use the type parameter `T`, there is no useful information about how the types of the two arguments should be related.

Also, the original file is actually incorrect here. The type of the `service` argument should be `ServiceDefinition`.

Reply:
What do you mean by the "specific definition"? The purpose of this change was to make the required type of implementation as specific as possible.

`{ [name: string]: handleCall }` basically means an object with any handlers, named in any way, so to me it is quite unspecific.

Reply:
The old definition says that the `implementation` type has to be an object that maps string names to one of four call handler function types. The new definition just says that the argument is `T`, which provides no information at all.